src/src/facilitator/facilitator.service.ts

```diff
@@ -2449,4 +2449,115 @@ export class FacilitatorService {
 			data: okyc_response || {},
 		});
 	}
+
+	public async createProgramFaciltator(body: any, request: any, res: any) {
+		let user_id = request?.mw_userid;
+		let academic_year_id = request?.mw_academic_year_id;
+
+		let { parent_ip, program_id } = body;
+
+		//validation to check if th faciltator is getting registered for the same program
+
+		let validation_query = `query MyQuery {
+			program_faciltators(where: {user_id: {_eq:${user_id}}}){
+			  program_id
+			  has_social_work_exp
+			  availability
+			  created_by
+			  documents_status
+			  eligibility_details
+			  eligibility_percentage
+			  form_step_number
+			  has_social_work_exp
+			  okyc_response
+			  police_verification_done
+			  qualification_ids
+			  status
+			  status_reason
+			  social_background_verified_by_neighbours
+			  village_knowledge_test
+			}
+		  }
+		  `;
+
+		const validation_result = await this.hasuraService.getData({
+			query: validation_query,
+		});
+
+		let program_faciltators = validation_result?.data?.program_faciltators;
+
+		const ids = program_faciltators.map(
+			(facilitator) => facilitator.program_id,
+		);
+
+		if (!ids.includes(body?.program_id)) {
+			return res.json({
+				status: 422,
+				success: false,
+				data: {},
+				message: 'Cannot add faciltator for another program',
+			});
+		}
+
+		//validation to check if the same faciltator data is present previously.
```

Comment:
Start code comments by giving one space after `//`
Start code comments in Capital case
eg:
		//start validation
=>
		// Start validations